contracts/strategy/LiquidationStrategy.sol

```diff
@@ -144,80 +163,140 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
-            idleWXDC.amountNeededToPayDebt -= deductAmountNeededToPayDebt;
-        } else {
-            idleWXDC.amountNeededToPayDebt = 0;
-        }
+    /// @param _permit2 The address of the UniswapV3 permit2 contract.
+    /// @param _universalRouter The address of the UniswapV3 UniversalRouter contract.
+    function setV3Info(address _permit2, address _universalRouter, uint24 _poolFee) external onlyStrategyManager {
+        if (_permit2 == address(0) || _universalRouter == address(0)) revert ZeroAddress();
+        if (_poolFee == 0) revert ZeroAmount();
+        if (uniswapV3Info[_universalRouter].permit2 == _permit2 && uniswapV3Info[_universalRouter].universalRouter == _universalRouter && uniswapV3Info[_universalRouter].poolFee == _poolFee) revert SameV3Info();
+        uniswapV3Info[_universalRouter] = UniswapV3Info({ permit2: _permit2, universalRouter: _universalRouter, poolFee: _poolFee });
+        emit LogSetV3Info(_permit2, _universalRouter);
+    }
 
-        WXDC.safeTransfer(strategyManager, _amount);
-        emit LogShutdownWithdrawWXDC(strategyManager, _amount);
+    function setUSDTInPath(address _usdToken) external onlyStrategyManager {
+        if (_usdToken == address(0)) revert ZeroAddress();
+        usdToken = ERC20(_usdToken);
+        emit LogSetUSDTInPath(_usdToken);
     }
 
-    /// @notice Allows the strategy manager to sell WXDC held by the contract.
+    /// @notice Allows the strategy manager to sell Collateral, held by the contract, to UniV2.
     /// @dev Only the current strategy manager can call this function.
-    /// This allows for managing the WXDC obtained through liquidations.
+    /// This allows for managing the Collateral obtained through liquidations.
     /// @param _router The UniswapV2Router02 (or a fork) used for the sale.
-    /// @param _amount The amount of WXDC to sell.
+    /// @param _amount The amount of Collateral to sell.
     /// @param _minAmountOut The minimum amount of FXD to accept for the sale.
-    function sellWXDC(IUniswapV2Router02 _router, uint256 _amount, uint256 _minAmountOut) external onlyStrategyManager {
-        require(address(_router) != address(0), "LiquidationStrategy: zero address");
-        require(_amount > 0, "LiquidationStrategy: zero amount");
-        require(_amount <= idleWXDC.WXDCAmount, "LiquidationStrategy: wrong amount");
+    function sellCollateralV2(address _collateral, IUniswapV2Router02 _router, uint256 _amount, uint256 _minAmountOut) external onlyStrategyManager {
+        if (address(_router) == address(0)) revert ZeroAddress();
+        if (_amount == 0) revert ZeroAmount();
+        if (_amount > idleCollateral[_collateral].CollateralAmount) revert WrongAmount();
 
-        idleWXDC.WXDCAmount -= _amount;
+        // Calculate the cost of Collateral in terms of FXD using the average price
+        uint256 costOfCollateralInFXD = _amount.mul(idleCollateral[_collateral].averagePriceOfCollateral).div(WAD); 
```

Comment:
we read `idleCollateral[_collateral].averagePriceOfCollateral` from the storage twice in this function, better to store data in local variable